modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/index/AbstractRocksDbIndexStorage.java

```diff
@@ -221,7 +221,16 @@ String createStorageInfo() {
      *
      * @throws RocksDBException If failed to delete data.
      */
-    public abstract void destroyData(WriteBatch writeBatch) throws RocksDBException;
+    public final void destroyData(WriteBatch writeBatch) throws RocksDBException {
+        destroyData0(writeBatch);
+
+        indexMetaStorage.removeNextRowIdToBuild(writeBatch, indexId, helper.partitionId());
+
+        nextRowIdToBuild = RowId.lowestRowId(helper.partitionId());
+    }
+
+    /** Methods that needs to be overridden by the inheritors to destroy implementation specific data. */
+    abstract void destroyData0(WriteBatch writeBatch) throws RocksDBException;
```

Comment:
Please give it a proper name. Names having `0` are already ugly, and it seems that we can only tolerate them as private methods. If it defines some kind of an interface, it should have a better name. How about `destroyIndexData()`?